Detectors/ITSMFT/ITS/simulation/src/V3Layer.cxx

```diff
@@ -0,0 +1,2384 @@
+const Double_t V3Layer::sOBHalfStaveYTrans = 1.76 * sMm;
+const Double_t V3Layer::sOBHalfStaveXOverlap = 4.3 * sMm;
+const Double_t V3Layer::sOBGraphiteFoilThick = 30.0 * sMicron;
+const Double_t V3Layer::sOBCoolTubeInnerD = 2.052 * sMm;
+const Double_t V3Layer::sOBCoolTubeThick = 32.0 * sMicron;
+const Double_t V3Layer::sOBCoolTubeXDist = 11.1 * sMm;
+
+const Double_t V3Layer::sOBSpaceFrameWidth = 42.0 * sMm;
+const Double_t V3Layer::sOBSpaceFrameTotHigh = 43.1 * sMm;
+const Double_t V3Layer::sOBSFrameBeamRadius = 0.6 * sMm;
+const Double_t V3Layer::sOBSpaceFrameLa = 3.0 * sMm;
+const Double_t V3Layer::sOBSpaceFrameHa = 0.721979 * sMm;
+const Double_t V3Layer::sOBSpaceFrameLb = 3.7 * sMm;
+const Double_t V3Layer::sOBSpaceFrameHb = 0.890428 * sMm;
+const Double_t V3Layer::sOBSpaceFrameL = 0.25 * sMm;
+const Double_t V3Layer::sOBSFBotBeamAngle = 56.5;
+const Double_t V3Layer::sOBSFrameBeamSidePhi = 65.0;
+
+ClassImp(V3Layer)
+
+#define SQ(A) (A) * (A)
+
+V3Layer::V3Layer()
+  : V11Geometry(),
+    mLayerNumber(0),
+    mPhi0(0),
+    mLayerRadius(0),
+    mZLength(0),
+    mSensorThickness(0),
+    mChipThickness(0),
+    mStaveWidth(0),
+    mStaveTilt(0),
+    mNumberOfStaves(0),
+    mNumberOfModules(0),
+    mNumberOfChips(0),
+    mChipTypeID(0),
+    mIsTurbo(0),
+    mBuildLevel(0),
+    mStaveModel(Detector::kIBModelDummy)
+{
+  for (int i = kNHLevels; i--;) {
+    mHierarchy[i] = 0;
+  }
+}
+
+V3Layer::V3Layer(Int_t debug)
+  : V11Geometry(debug),
+    mLayerNumber(0),
+    mPhi0(0),
+    mLayerRadius(0),
+    mZLength(0),
+    mSensorThickness(0),
+    mChipThickness(0),
+    mStaveWidth(0),
+    mStaveTilt(0),
+    mNumberOfStaves(0),
+    mNumberOfModules(0),
+    mNumberOfChips(0),
+    mChipTypeID(0),
+    mIsTurbo(0),
+    mBuildLevel(0),
+    mStaveModel(Detector::kIBModelDummy)
+{
+  for (int i = kNHLevels; i--;) {
+    mHierarchy[i] = 0;
+  }
+}
+
+V3Layer::V3Layer(Int_t lay, Int_t debug)
+  : V11Geometry(debug),
+    mLayerNumber(lay),
+    mPhi0(0),
+    mLayerRadius(0),
+    mZLength(0),
+    mSensorThickness(0),
+    mChipThickness(0),
+    mStaveWidth(0),
+    mStaveTilt(0),
+    mNumberOfStaves(0),
+    mNumberOfModules(0),
+    mNumberOfChips(0),
+    mChipTypeID(0),
+    mIsTurbo(0),
+    mBuildLevel(0),
+    mStaveModel(Detector::kIBModelDummy)
+{
+  for (int i = kNHLevels; i--;) {
+    mHierarchy[i] = 0;
+  }
+}
+
+V3Layer::V3Layer(Int_t lay, Bool_t turbo, Int_t debug)
+  : V11Geometry(debug),
+    mLayerNumber(lay),
+    mPhi0(0),
+    mLayerRadius(0),
+    mZLength(0),
+    mSensorThickness(0),
+    mChipThickness(0),
+    mStaveWidth(0),
+    mStaveTilt(0),
+    mNumberOfStaves(0),
+    mNumberOfModules(0),
+    mNumberOfChips(0),
+    mChipTypeID(0),
+    mIsTurbo(turbo),
+    mBuildLevel(0),
+    mStaveModel(Detector::kIBModelDummy)
+{
+  for (int i = kNHLevels; i--;) {
+    mHierarchy[i] = 0;
+  }
+}
+
+V3Layer::V3Layer(const V3Layer &s)
+  : V11Geometry(s.getDebug()),
```

Comment:
I got rid of my implementation of copy and assignment constructors, and used "default" in V3Layer.h, please let me know if it is ok.